### DISCUSSION TITLE
TcpSocket drop question

### DISCUSSION THREAD

**Discussion by panpur** [tokio-rs/tokio#3724] [Q&A] [answered]
Hello,

As per the [documentation](https://docs.rs/tokio/1.5.0/tokio/net/struct.TcpSocket.html) 
> The underlying socket is closed when the TcpSocket value is dropped.

If I call `drop(socket)`, is socket closed synchronously or is it occurred on other thread?
The reason is, I'm working on app to measure performance of network operation.
One of the KPI is disconnect time.

Is this achievable?
If it's closed synchronously I suppose I can do something like this: 
`let disconnect_time = Instant::now();`
`drop(tcp);`
`result.disconnect_time = Some(disconnect_time.elapsed().as_micros());`

Thanks

**Comment by Darksonn** [comment, marked as answer]
Yes, a tcp connection will close its file descriptor synchronously. The only type that doesn't do this is `File`, where it may not be closed immediately if there are any in-flight operations.

**Reply by panpur** [reply]
superb!
